air/src/air/mod.rs

```diff
@@ -188,6 +189,12 @@ pub trait Air: Send + Sync {
     /// This could be any type as long as it can be serialized into a sequence of field elements.
     type PublicInputs: ToElements<Self::BaseField>;
 
+    /// A type defining the random elements used in constructing the auxiliary trace segment.
+    type AuxRandElements<E: Send + Sync>: Send + Sync;
```

Comment:
I would probably also impose a trait bound on `E: FieldElement<BaseField = Self::BaseField>`.

Reply:
Fixed, but I think we should move away from adding trait bounds in type definitions, as [discussed on Discord](https://discord.com/channels/893151281848406016/1166735622455689268/1167139113024098364) (pointing to this [StackOverflow thread](https://stackoverflow.com/questions/49229332/should-trait-bounds-be-duplicated-in-struct-and-impl/66369912#66369912))

Reply:
Agreed - but we should probably do it more comprehensively in a follow-up PR.